src/tools/illink/src/linker/Linker/TypeMapInfo.cs

```diff
@@ -313,8 +311,7 @@ IEnumerable<InterfaceImplementation> GetDefaultInterfaceImplementations (TypeDef
 				// We haven't found a MethodImpl on the current interface, but one of the interfaces
 				// this interface requires could still provide it.
 				if (!foundImpl) {
-					foreach (var impl in GetDefaultInterfaceImplementations (potentialImplInterface, interfaceMethod))
-						yield return impl;
+					FindAndAddDefaultInterfaceImplementations (potentialImplInterface, interfaceMethod);
```

Comment:
This is not producing equivalent results.

Previously, we always called `AddDefaultInterfaceImplementation (resolvedInterfaceMethod, type, {ThisComesFromTheAlgorithm});`

Now we are calling `AddDefaultInterfaceImplementation (resolvedInterfaceMethod, {ThisComesFromTheAlgorithm}, {ThisComesFromTheAlgorithm});` because at this spot `type` is swapped for `potentialImplInterface` when this recursively calls itself at this spot.

Is that a problem? Or the second parameter doesn't matter for anything? If it doesn't matter, the perf improvement in #98436 can be even better - we don't have to look for default implementation per each class - we can only look for default implementations when `MapType` is called for an interface and don't worry about default implementations when looking at classes at all.

Reply:
Oh, good catch. We do need the correct 'ImplementingType' for this, so we'll have to make sure the original `type` is passed to `AddDefaultInterfaceImplementation`.